src/components/v5/common/Pills/PillsBase.tsx

```diff
@@ -14,33 +14,39 @@ const PillsBase: FC<PropsWithChildren<PillsProps>> = ({
   isCapitalized = true,
   textClassName,
   ...rest
-}) => (
-  <span
-    className={clsx(
-      'inline-flex shrink-0 items-center rounded-3xl px-3 py-1 text-center',
-      className,
-      {
-        capitalize: isCapitalized,
-        'h-[1.625rem] text-3': pillSize === 'medium',
-        'h-[1.5rem] text-4': pillSize === 'small',
-      },
-    )}
-    {...rest}
-  >
-    {Icon && (
-      <span className="flex shrink-0">
-        <Icon size={pillSize === 'medium' ? 14 : 12} />
-      </span>
-    )}
+}) => {
+  const content = text || children;
+
+  return (
     <span
-      className={clsx(textClassName, {
-        'ml-1': !!Icon,
-      })}
+      className={clsx(
+        'inline-flex shrink-0 items-center rounded-3xl px-2.5 py-1.5 text-center',
+        className,
+        {
+          capitalize: isCapitalized,
+          'h-[1.625rem] text-3': pillSize === 'medium',
+          'h-[1.5rem] text-4': pillSize === 'small',
+        },
+      )}
+      {...rest}
```

Comment:
Hi there, thanks for your PR! Just a quick note: I noticed something about props spreading and types that we could improve. There are some props in the type that aren't being extracted from props, which means they might end up as attributes in the <span> element by accident.

Here are the props that could be spread into <span>:

```
export interface PillsProps {
  mode?: ExtensionStatusBadgeMode | UserStatusMode;
  iconSize?: IconSize;
  isIconVisible?: boolean;
}
```